pkg/xtrabackup/xtrabackup.go

```diff
@@ -0,0 +1,386 @@
+						xtrabackupContainer(cluster, cr.Name, destination, storage),
+					},
+					SecurityContext:   storage.PodSecurityContext,
+					Affinity:          storage.Affinity,
+					Tolerations:       storage.Tolerations,
+					NodeSelector:      storage.NodeSelector,
+					SchedulerName:     storage.SchedulerName,
+					PriorityClassName: storage.PriorityClassName,
+					RuntimeClassName:  storage.RuntimeClassName,
+					DNSPolicy:         corev1.DNSClusterFirst,
+					Volumes: []corev1.Volume{
+						{
+							Name: dataVolumeName,
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+						{
+							Name: credsVolumeName,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: cluster.Spec.SecretsName,
+								},
+							},
+						},
+						{
+							Name: tlsVolumeName,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: cluster.Spec.SSLSecretName,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func xtrabackupContainer(cluster *apiv1alpha1.PerconaServerMySQL, backupName, destination string, storage *apiv1alpha1.BackupStorageSpec) corev1.Container {
+	spec := cluster.Spec.Backup
+
+	verifyTLS := true
+	if storage.VerifyTLS != nil {
+		verifyTLS = *storage.VerifyTLS
+	}
+
+	return corev1.Container{
+		Name:            componentName,
+		Image:           spec.Image,
+		ImagePullPolicy: spec.ImagePullPolicy,
+		Env: []corev1.EnvVar{
+			{
+				Name:  "BACKUP_NAME",
+				Value: backupName,
+			},
+			{
+				Name:  "BACKUP_DEST",
+				Value: destination,
+			},
+			{
+				Name:  "VERIFY_TLS",
+				Value: strconv.FormatBool(verifyTLS),
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{
+				Name:      dataVolumeName,
+				MountPath: dataMountPath,
+			},
+			{
+				Name:      tlsVolumeName,
+				MountPath: tlsMountPath,
+			},
+		},
+		Command:                  []string{"/var/lib/mysql/run-backup.sh"},
+		TerminationMessagePath:   "/dev/termination-log",
+		TerminationMessagePolicy: corev1.TerminationMessageReadFile,
+		SecurityContext:          storage.ContainerSecurityContext,
+		Resources:                storage.Resources,
+	}
+}
+
+func PVC(cluster *apiv1alpha1.PerconaServerMySQL, cr *apiv1alpha1.PerconaServerMySQLBackup, storage *apiv1alpha1.BackupStorageSpec) *corev1.PersistentVolumeClaim {
+	if len(storage.Volume.PersistentVolumeClaim.AccessModes) == 0 {
+		storage.Volume.PersistentVolumeClaim.AccessModes = []corev1.PersistentVolumeAccessMode{
+			corev1.ReadWriteOnce,
+		}
+	}
+
+	return &corev1.PersistentVolumeClaim{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "batch/v1",
+			Kind:       "Job",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      JobName(cr),
+			Namespace: cluster.Namespace,
+		},
+		Spec: *storage.Volume.PersistentVolumeClaim,
+	}
+}
+
+func SetStoragePVC(job *batchv1.Job, pvc *corev1.PersistentVolumeClaim) error {
```

Comment:
the same for this fun

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/77#discussion_r901343924